internal/pkg/otel/manager/manager.go

```diff
@@ -361,9 +362,30 @@ func buildMergedConfig(cfgUpdate configUpdate, agentInfo info.Agent, monitoringC
 		}
 	}
 
+	if err := m.injectDiagnosticsExtension(mergedOtelCfg); err != nil {
+		return nil, fmt.Errorf("failed to inject diagnostics: %w", err)
+	}
+
 	return mergedOtelCfg, nil
 }
 
+func (m *OTelManager) injectDiagnosticsExtension(config *confmap.Conf) error {
```

Comment:
This doesn't need to be a manager method, as it doesn't use the manager at all. Nor does `buildMergedConfig` above need to be one. Imo it's better to use pure functions where feasible, as it's easier to understand what data they're actually operating on.